src/gluonts/core/serde/_base.py

```diff
@@ -309,13 +309,13 @@ def decode(r: Any) -> Any:
     """
 
     # structural recursion over the possible shapes of r
-    if type(r) == dict and "__kind__" in r:
+    if isinstance(r, dict) and "__kind__" in r:
         kind = r["__kind__"]
         cls = cast(Any, locate(r["class"]))
 
         assert cls is not None, f"Can not locate {r['class']}."
 
-        if kind == Kind.Type:
+        if kind == Kind.Type.value:
```

Comment:
I think this is not needed

Reply:
If I understand correctly, this line as it was originally, i.e. `if kind == Kind.Type:` is not passing the style checks. Any suggestions on how to fix it?

Reply:
I think it's passing them, is just a comparison between strings as far as I understand